### ISSUE TITLE
MySQL does not start with overlay2 and overlay storage driver

### ISSUE THREAD

**Issue by felixlinker** [KaiHofstetter/docker-wordpress-cli#3]
I tried to set up wordpress following your instructions. This is what I type in to cmdline:

    C:\Git\docker-wordpress-cli>docker run --name wp-dev -p 80:80 -p 3306:3306 -d kaihofstetter/wordpress-cli
    487a731a4d422d7da07d00718a5e6a772ffd50b6314ac089872b5bec67fbbef9

    C:\Git\docker-wordpress-cli>docker logs wp-dev
    => MySQL not configured yet, configuring MySQL ...
    => Setting MySQL bind address to have access from the Docker host
    => Starting MySQL
    => Waiting till MySQL is started
     * Starting MySQL database server mysqld
       ...fail!
    => Creating WordPress db
    => Creating MySQL WP user and granting external access
    mysqladmin: connect to server at 'localhost' failed
        error: 'Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)'
        Check that mysqld is running and that the socket: '/var/run/mysqld/mysqld.sock' exists!
    ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
    ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
    => Apache is not configured yet, configuring Apache ...
    => Setting ServerName
    Error: Error establishing a database connection. This either means that the username and password information in your `wp-config.php` file is incorrect or we can’t contact the database server at `localhost`. This could mean your host’s database server is down.
    => WordPress is not configured yet, configuring WordPress ...
    => Installing WordPress
    Error: Error establishing a database connection. This either means that the username and password information in your `wp-config.php` file is incorrect or we can’t contact the database server at `localhost`. This could mean your host’s database server is down.
    => Apache started...

As to be expected, trying to connect to worpress results in **`Error establishing a database connection`** and the status code `500`.

Here are the results from `docker info`:

    C:\Git\docker-wordpress-cli>docker info
    Containers: 1
     Running: 1
     Paused: 0
     Stopped: 0
    Images: 8
    Server Version: 17.12.0-ce
    Storage Driver: overlay2
     Backing Filesystem: extfs
     Supports d_type: true
     Native Overlay Diff: true
    Logging Driver: json-file
    Cgroup Driver: cgroupfs
    Plugins:
     Volume: local
     Network: bridge host ipvlan macvlan null overlay
     Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
    Swarm: inactive
    Runtimes: runc
    Default Runtime: runc
    Init Binary: docker-init
    containerd version: 89623f28b87a6004d4b785663257362d1658a729
    runc version: b2567b37d7b75eb4cf325b77297b140ea686ce8f
    init version: 949e6fa
    Security Options:
     seccomp
      Profile: default
    Kernel Version: 4.9.60-linuxkit-aufs
    Operating System: Docker for Windows
    OSType: linux
    Architecture: x86_64
    CPUs: 2
    Total Memory: 1.934GiB
    Name: linuxkit-00155d66cb05
    ID: 36YU:MJO2:V66C:65TR:7PQK:FDWR:QLAL:AEI2:MN6J:LBSE:2Q6F:CEG7
    Docker Root Dir: /var/lib/docker
    Debug Mode (client): false
    Debug Mode (server): true
     File Descriptors: 26
     Goroutines: 44
     System Time: 2018-01-17T18:37:48.8666809Z
     EventsListeners: 1
    Registry: https://index.docker.io/v1/
    Labels:
    Experimental: true
    Insecure Registries:
     127.0.0.0/8
    Live Restore Enabled: false

**Comment by KaiHofstetter** [comment]
I just tried to reproduce the error with the exact command you used and freshly downloaded docker images, but it works on my machine.

Could you start a bash in the docker container:
`sudo docker exec -it wp-dev bash`

The MySQL logs in the container might give us more info about the error:
`vi var/log/mysql/error.log`

**Comment by felixlinker** [comment]
Here's tone content of the `error.log` file:

    171130 21:13:28 [Warning] Using unique option prefix myisam-recover instead of mm
    yisam-recover-options is deprecated and will be removed in a future release. Plee
    ase use the full name instead.
    171130 21:13:28 [Note] Plugin 'FEDERATED' is disabled.
    171130 21:13:28 InnoDB: The InnoDB memory heap is disabled
    171130 21:13:28 InnoDB: Mutexes and rw_locks use GCC atomic builtins
    171130 21:13:28 InnoDB: Compressed tables use zlib 1.2.8
    171130 21:13:28 InnoDB: Using Linux native AIO
    171130 21:13:28 InnoDB: Initializing buffer pool, size = 128.0M
    171130 21:13:28 InnoDB: Completed initialization of buffer pool
    InnoDB: The first specified data file ./ibdata1 did not exist:
    InnoDB: a new database to be created!
    171130 21:13:28  InnoDB: Setting file ./ibdata1 size to 10 MB
    InnoDB: Database physically writes the file full: wait...
    171130 21:13:29  InnoDB: Log file ./ib_logfile0 did not exist: new to be created
    InnoDB: Setting log file ./ib_logfile0 size to 5 MB
    InnoDB: Database physically writes the file full: wait...
    171130 21:13:29  InnoDB: Log file ./ib_logfile1 did not exist: new to be created
    InnoDB: Setting log file ./ib_logfile1 size to 5 MB
    InnoDB: Database physically writes the file full: wait...
    InnoDB: Doublewrite buffer not found: creating new
    InnoDB: Doublewrite buffer created
    InnoDB: 127 rollback segment(s) active.
    "var/log/mysql/error.log" 93 lines, 6223 characters

**Comment by KaiHofstetter** [comment]
This is the beginning of the log file. When you scroll down, there should be log entries with the current date, like:

````
...
180118 19:13:36  InnoDB: Waiting for the background threads to start
180118 19:13:37 InnoDB: 5.5.58 started; log sequence number 1595675
180118 19:13:37 [Note] Server hostname (bind-address): '0.0.0.0'; port: 3306
180118 19:13:37 [Note]   - '0.0.0.0' resolves to '0.0.0.0';
180118 19:13:37 [Note] Server socket created on IP: '0.0.0.0'.
180118 19:13:37 [Note] Event Scheduler: Loaded 0 events
180118 19:13:37 [Note] /usr/sbin/mysqld: ready for connections.
Version: '5.5.58-0ubuntu0.14.04.1'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  (Ubuntu)
````

**Comment by felixlinker** [comment]
Sorry, didn't thought about that. Here's the complete file:

```
root@4d322ea6ae37:/# cat var/log/mysql/error.log
171130 21:13:28 [Warning] Using unique option prefix myisam-recover instead of myisam-recover-options is deprecated and will be removed in a future release. Please use the full name instead.
171130 21:13:28 [Note] Plugin 'FEDERATED' is disabled.
171130 21:13:28 InnoDB: The InnoDB memory heap is disabled
171130 21:13:28 InnoDB: Mutexes and rw_locks use GCC atomic builtins
171130 21:13:28 InnoDB: Compressed tables use zlib 1.2.8
171130 21:13:28 InnoDB: Using Linux native AIO
171130 21:13:28 InnoDB: Initializing buffer pool, size = 128.0M
171130 21:13:28 InnoDB: Completed initialization of buffer pool
InnoDB: The first specified data file ./ibdata1 did not exist:
InnoDB: a new database to be created!
171130 21:13:28  InnoDB: Setting file ./ibdata1 size to 10 MB
InnoDB: Database physically writes the file full: wait...
171130 21:13:29  InnoDB: Log file ./ib_logfile0 did not exist: new to be created
InnoDB: Setting log file ./ib_logfile0 size to 5 MB
InnoDB: Database physically writes the file full: wait...
171130 21:13:29  InnoDB: Log file ./ib_logfile1 did not exist: new to be created
InnoDB: Setting log file ./ib_logfile1 size to 5 MB
InnoDB: Database physically writes the file full: wait...
InnoDB: Doublewrite buffer not found: creating new
InnoDB: Doublewrite buffer created
InnoDB: 127 rollback segment(s) active.
InnoDB: Creating foreign key constraint system tables
InnoDB: Foreign key constraint system tables created
171130 21:13:29  InnoDB: Waiting for the background threads to start
171130 21:13:30 InnoDB: 5.5.58 started; log sequence number 0
171130 21:13:30  InnoDB: Starting shutdown...
171130 21:13:31  InnoDB: Shutdown completed; log sequence number 1595675
171130 21:13:31 [Warning] Using unique option prefix myisam-recover instead of myisam-recover-options is deprecated and will be removed in a future release. Please use the full name instead.
171130 21:13:31 [Note] Plugin 'FEDERATED' is disabled.
171130 21:13:31 InnoDB: The InnoDB memory heap is disabled
171130 21:13:31 InnoDB: Mutexes and rw_locks use GCC atomic builtins
171130 21:13:31 InnoDB: Compressed tables use zlib 1.2.8
171130 21:13:31 InnoDB: Using Linux native AIO
171130 21:13:31 InnoDB: Initializing buffer pool, size = 128.0M
171130 21:13:31 InnoDB: Completed initialization of buffer pool
171130 21:13:32 InnoDB: highest supported file format is Barracuda.
171130 21:13:32  InnoDB: Waiting for the background threads to start
171130 21:13:33 InnoDB: 5.5.58 started; log sequence number 1595675
ERROR: 1064  You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'ALTER TABLE user ADD column Show_view_priv enum('N','Y') CHARACTER SET utf8 NOT ' at line 1
171130 21:13:33 [ERROR] Aborting

171130 21:13:33  InnoDB: Starting shutdown...
171130 21:13:33  InnoDB: Shutdown completed; log sequence number 1595675
171130 21:13:33 [Note] /usr/sbin/mysqld: Shutdown complete

171130 21:13:33 [Warning] Using unique option prefix myisam-recover instead of myisam-recover-options is deprecated and will be removed in a future release. Please use the full name instead.
171130 21:13:33 [Note] Plugin 'FEDERATED' is disabled.
171130 21:13:33 InnoDB: The InnoDB memory heap is disabled
171130 21:13:33 InnoDB: Mutexes and rw_locks use GCC atomic builtins
171130 21:13:33 InnoDB: Compressed tables use zlib 1.2.8
171130 21:13:33 InnoDB: Using Linux native AIO
171130 21:13:34 InnoDB: Initializing buffer pool, size = 128.0M
171130 21:13:34 InnoDB: Completed initialization of buffer pool
171130 21:13:34 InnoDB: highest supported file format is Barracuda.
171130 21:13:34  InnoDB: Waiting for the background threads to start
171130 21:13:35 InnoDB: 5.5.58 started; log sequence number 1595675
171130 21:13:35  InnoDB: Starting shutdown...
171130 21:13:35  InnoDB: Shutdown completed; log sequence number 1595675
171130 21:13:36 [Warning] Using unique option prefix myisam-recover instead of myisam-recover-options is deprecated and will be removed in a future release. Please use the full name instead.
171130 21:13:36 [Note] Plugin 'FEDERATED' is disabled.
171130 21:13:36 InnoDB: The InnoDB memory heap is disabled
171130 21:13:36 InnoDB: Mutexes and rw_locks use GCC atomic builtins
171130 21:13:36 InnoDB: Compressed tables use zlib 1.2.8
171130 21:13:36 InnoDB: Using Linux native AIO
171130 21:13:36 InnoDB: Initializing buffer pool, size = 128.0M
171130 21:13:36 InnoDB: Completed initialization of buffer pool
171130 21:13:36 InnoDB: highest supported file format is Barracuda.
171130 21:13:36  InnoDB: Waiting for the background threads to start
171130 21:13:37 InnoDB: 5.5.58 started; log sequence number 1595675
ERROR: 1050  Table 'plugin' already exists
171130 21:13:37 [ERROR] Aborting

171130 21:13:37  InnoDB: Starting shutdown...
171130 21:13:38  InnoDB: Shutdown completed; log sequence number 1595675
171130 21:13:38 [Note] /usr/sbin/mysqld: Shutdown complete

180119 13:25:58 [Warning] Using unique option prefix myisam-recover instead of myisam-recover-options is deprecated and will be removed in a future release. Please use the full name instead.
180119 13:25:58 [Note] Plugin 'FEDERATED' is disabled.
/usr/sbin/mysqld: Got error 140 from storage engine
180119 13:25:58 [ERROR] Can't open the mysql.plugin table. Please run mysql_upgrade to create it.
180119 13:25:58 InnoDB: The InnoDB memory heap is disabled
180119 13:25:58 InnoDB: Mutexes and rw_locks use GCC atomic builtins
180119 13:25:58 InnoDB: Compressed tables use zlib 1.2.8
180119 13:25:58 InnoDB: Using Linux native AIO
180119 13:25:58 InnoDB: Initializing buffer pool, size = 128.0M
180119 13:25:58 InnoDB: Completed initialization of buffer pool
180119 13:25:58 InnoDB: highest supported file format is Barracuda.
180119 13:25:58  InnoDB: Waiting for the background threads to start
180119 13:25:59 InnoDB: 5.5.58 started; log sequence number 1595675
180119 13:25:59 [Note] Server hostname (bind-address): '0.0.0.0'; port: 3306
180119 13:25:59 [Note]   - '0.0.0.0' resolves to '0.0.0.0';
180119 13:25:59 [Note] Server socket created on IP: '0.0.0.0'.
180119 13:25:59 [ERROR] Fatal error: Can't open and lock privilege tables: Got error 140 from storage engine
```

**Comment by KaiHofstetter** [comment]
It seems that you ran into this problem:
https://github.com/docker/for-linux/issues/72

I'll investigate that further...

Thank you for your feedback!
